### PR TITLE
Add `databricks_mws_credentials` data source

### PR DESCRIPTION
**Pull request by boydZD**
Lists all [databricks_mws_credentials](../resources/mws_credentials.md) in Databricks Account.

-> **Note** [`account_id`](../index.md#account_id) provider configuration property is required for this resource to work.

## Example Usage

Listing all workspaces in 

```hcl
provider "databricks" {
  // other configuration
  account_id = "<databricks account id>"
}

data "databricks_mws_credentials" "all" {}

output "all_mws_credentials" {
  value = data.databricks_mws_credentials.all.ids
}
```

## Attribute Reference

-> **Note** This resource has an evolving interface, which may change in future versions of the provider.

This data source exports the following attributes:

* `ids` - name-to-id map for all of the credentials in the account

## Related Resources

The following resources are used in the same context:

* [databricks_mws_workspaces](../resources/mws_workspaces.md) to manage Databricks E2 Workspaces.
* [databricks_metastore_assignment](../resources/metastore_assignment.md) to assign [databricks_metastore](docs/resources/metastore.md) to [databricks_mws_workspaces](../resources/mws_workspaces.md) or [azurerm_databricks_workspace](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/databricks_workspace)

### CODE REVIEW

File: docs/data-sources/mws_credentials.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+subcategory: "AWS"
+---
+# databricks_mws_credentials Data Source
+
+-> **Note** If you have a fully automated setup with workspaces created by [databricks_mws_workspaces](../resources/mws_workspaces.md) or [azurerm_databricks_workspace](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/databricks_workspace), please make sure to add [depends_on attribute](../index.md#data-resources-and-authentication-is-not-configured-errors) in order to prevent _authentication is not configured for provider_ errors.
```
Comment (alexott) [inline docs/data-sources/mws_credentials.md:6]: in this context `azurerm_databricks_workspace` isn't relevant

File: docs/data-sources/mws_credentials.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+This data source exports the following attributes:
+
+* `ids` - name-to-id map for all of the credentials in the account
+
+## Related Resources
+
+The following resources are used in the same context:
+
+* [databricks_mws_workspaces](../resources/mws_workspaces.md) to manage Databricks E2 Workspaces.
+* [databricks_metastore_assignment](../resources/metastore_assignment.md) to assign [databricks_metastore](docs/resources/metastore.md) to [databricks_mws_workspaces](../resources/mws_workspaces.md) or [azurerm_databricks_workspace](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/databricks_workspace)
```
Comment (alexott) [inline docs/data-sources/mws_credentials.md:42]: is Unity Catalog relevant here? Plus same comment about azure workspace...

File: mws/data_mws_credentials.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	"context"
+	"fmt"
+
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceMwsCredentials() *schema.Resource {
+	type mwsCredentialsData struct {
+		Ids map[string]string `json:"ids,omitempty" tf:"computed,slice_set"`
```
Comment (alexott) [inline mws/data_mws_credentials.go:13]: `slice_set` isn't necessary here. It's only for lists

**Review by alexott — CHANGES_REQUESTED** [review]
Just small changes required, otherwise looks good

File: docs/data-sources/mws_credentials.md (hunk 1 above)
Comment (boydZD) [reply docs/data-sources/mws_credentials.md]: addressed in latest commit

File: docs/data-sources/mws_credentials.md (hunk 2 above)
Comment (boydZD) [reply docs/data-sources/mws_credentials.md]: addressed in latest commit

File: mws/data_mws_credentials.go (hunk 3 above)
Comment (boydZD) [reply mws/data_mws_credentials.go]: Removed slice_set

File: mws/data_mws_credentials.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+func DataSourceMwsCredentials() *schema.Resource {
+	type mwsCredentialsData struct {
+		Ids map[string]string `json:"ids,omitempty" tf:"computed"`
+	}
+	return common.DataResource(mwsCredentialsData{}, func(ctx context.Context, e any, c *common.DatabricksClient) error {
+		data := e.(*mwsCredentialsData)
+		if c.AccountID == "" {
+			return fmt.Errorf("provider block is missing `account_id` property")
```
Comment (alexott) [inline mws/data_mws_credentials.go:18]: This execution path isn't tested. Test is simple:

```
unc TestDataSourceMwsCredentials(t *testing.T) {
	qa.ResourceFixture{
		Fixtures: []qa.HTTPFixture{},
		Resource:    DataSourceMwsCredentials(),
		Read:        true,
		NonWritable: true,
		ID:          "_",
	}.ExpectError("provider block is missing `account_id` property")
}
```

File: mws/data_mws_credentials.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+		}
+		credentials, err := NewCredentialsAPI(ctx, c).List(c.AccountID)
+		if err != nil {
+			return err
```
Comment (alexott) [inline mws/data_mws_credentials.go:22]: This as well isn't tested. Test could be simple:

```
func TestDataSourceMwsCredentials_Error(t *testing.T) {
	qa.ResourceFixture{
		Fixtures: qa.HTTPFailures,
		AccountID:   "abc",
		Resource:    DataSourceMwsCredentials(),
		Read:        true,
		NonWritable: true,
		ID:          "_",
	}.ExpectError(t, "I'm a teapot")
```

**Review by alexott — CHANGES_REQUESTED** [review]
let increase test code coverage

File: mws/data_mws_credentials.go (hunk 4 above)
Comment (boydZD) [reply mws/data_mws_credentials.go:18]: Unit tests updated as instructed. coverage 100%

**Review by nfx — APPROVED** [review]

**Pull request merged**